projects/dotcms-field-elements/src/components/dot-textfield-1/dot-textfield-1.tsx

```diff
@@ -0,0 +1,187 @@
+import { Component, Prop, State, Event, EventEmitter, Method } from '@stencil/core';
+import Fragment from 'stencil-fragment';
+import { DotFieldStatus } from '../../models/dot-field-status.model';
+
+interface DotFieldEvent {
+    name: string;
+}
+
+interface DotFieldStatusEvent extends DotFieldEvent {
+    status: DotFieldStatus;
+}
+
+interface DotFieldValueEvent extends DotFieldEvent {
+    value: string;
+}
+
+const getOriginalStatus = (): DotFieldStatus => {
+    return {
+        dotValid: true,
+        dotTouched: false,
+        dotPristine: true
+    };
+};
+
+const updateStatus = (
+    state: DotFieldStatus,
+    change: { [key: string]: boolean }
+): DotFieldStatus => {
+    return {
+        ...state,
+        ...change
+    };
+};
+
+const getClassNames = (status: DotFieldStatus, isValid: boolean) => {
+    return {
+        'dot-valid': isValid,
+        'dot-invalid': !isValid,
+        'dot-pristine': status.dotPristine,
+        'dot-dirty': !status.dotPristine,
+        'dot-touched': status.dotTouched,
+        'dot-untouched': !status.dotTouched
+    };
+};
+
+const getTagHint = (hint: string): string => {
+    return hint ? <span class='dot-field__hint'>{this.hint}</span> : '';
+};
+
+const getTagError = (show: boolean, message: string): string => {
+    return show ? <span class='dot-field__error-meessage'>{message}</span> : '';
+};
+
+const getTagLabel = (name: string, label: string): string => {
+    return <label htmlFor={name}>{label}</label>;
+};
+
+const getErrorClass = (valid: boolean): string => {
+    return valid ? '' : 'dot-field__input--error';
+};
+
```

Comment:
AFAIK, from line 1 to here, all this will go to the `Utils` class?

if so, prob we can remove `@Event() valueChange: EventEmitter<DotFieldValueEvent>;` and `@Event() statusChange: EventEmitter<DotFieldStatusEvent>;` from each field and also the `emits`, so in the end the `util` class will have a method to emit from there

Reply:
But how do we pass the `this.statusChange` and `this.valueChange` to that function?

Reply:
I guess you can pass the `JSON Object` (based on type), and then the util function will only do the emit of the JS object